### DISCUSSION TITLE
Why does t macro convert escaped newline (\n) to escaped backslash+n (\\n)

### DISCUSSION THREAD

**Discussion by sbixby** [lingui/js-lingui#1845] [Q&A]
I have a significant number of messages in functions in my React modules  that have newlines; the strings are passed to a custom dialog to control the message flow onscreen.

For example:

```
  const someFunc = () => {
    if (somesituation) {
      setAlert({
        open: true,
        message: t`some message\nwith two lines`
      })
    }
  }
```

This extracts into a po file as:

```
msgid "some message\\nwith two lines"
msgstr "some message\\nwith two lines"
```  

I *can* use template strings' multi-line, but then I give up clean code formatting where the template strings go to the first column.

What I've been doing is creating a msg template, eg:

```
  const someMsg = msg({
    id: 'somemsg',
    message: 'some message\nwith two lines'
  })
  

  const someFunc = () => {
    if (somesituation) {
      setAlert({
        open: true,
        message: i18n._(someMsg);
      })
    }
  }
```
Which works, but it's ... verbose.     

Am I missing something?   Is this a feature, a bug, etc?

**Comment by timofei-iatsenko** [comment]
`msg` and `t` has the same source code, `t` == `i18n._(msg())`. So the result should be the same. So you can write this inline: 

```ts
const someFunc = () => {
  if (somesituation) {
    setAlert({
      open: true,
      message: t({
          id: 'somemsg',
          message: 'some message\nwith two lines'
        });
    })
  }
}
```

I will re-check the case with escaped new line sequences. THis escaping actually done by babel / swc when the code is converted to AST but when it printed back by macro it should be correctly processed as new line without extra escaping....

**Reply by sbixby** [reply]
Thank you for that example, I guess I've done that as well but over time I was trying a number of different things to get past the t`` \\n thing, kind of forgot the basic one. :)
